internal/dump/dump.go

```diff
@@ -0,0 +1,40 @@
+package dump
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-redis/redis/v8"
+	"go.riyazali.net/sqlite"
+)
+
+type dump struct {
+	rdb *redis.Client
+}
+
+func (f *dump) Args() int           { return -1 }
+func (f *dump) Deterministic() bool { return false }
+func (f *dump) Apply(ctx *sqlite.Context, values ...sqlite.Value) {
+	var (
+		key    string
+		result *redis.StringCmd
+		err    error
+	)
+	if len(values) == 1 {
+		key = values[0].Text()
+
+	} else {
+		err = fmt.Errorf("must input single arg to dump")
+		ctx.ResultError(err)
+		return
+	}
+	result = f.rdb.Dump(context.TODO(), key)
```

Comment:
```suggestion
	result, err := f.rdb.Dump(context.TODO(), key).Result()
	if err != nil {
	  ctx.ResultError(err)
	  return
	}
```

two changes here - the `:=` so there's no need to declare `result` up above, and calling [`.Result()`](https://pkg.go.dev/github.com/go-redis/redis/v8#StringCmd.Result) on the return value should get you the string output of the Redis command (and handle the err)